tools/traffic/workers/blob_reader.go

```diff
@@ -0,0 +1,238 @@
+	logger logging.Logger,
+	config *config.WorkerConfig,
+	retriever clients.RetrievalClient,
+	chainClient eth.ChainClient,
+	blobStore *table.BlobStore,
+	generatorMetrics metrics.Metrics) BlobReader {
+
+	return BlobReader{
+		ctx:         ctx,
+		waitGroup:   waitGroup,
+		logger:      logger,
+		config:      config,
+		retriever:   retriever,
+		chainClient: chainClient,
+		blobsToRead: blobStore,
+		metrics: &blobReaderMetrics{
+			generatorMetrics:           generatorMetrics,
+			fetchBatchHeaderMetric:     generatorMetrics.NewLatencyMetric("fetch_batch_header"),
+			fetchBatchHeaderSuccess:    generatorMetrics.NewCountMetric("fetch_batch_header_success"),
+			fetchBatchHeaderFailure:    generatorMetrics.NewCountMetric("fetch_batch_header_failure"),
+			recombinationSuccessMetric: generatorMetrics.NewCountMetric("recombination_success"),
+			recombinationFailureMetric: generatorMetrics.NewCountMetric("recombination_failure"),
+			readLatencyMetric:          generatorMetrics.NewLatencyMetric("read"),
+			validBlobMetric:            generatorMetrics.NewCountMetric("valid_blob"),
+			invalidBlobMetric:          generatorMetrics.NewCountMetric("invalid_blob"),
+			readSuccessMetric:          generatorMetrics.NewCountMetric("read_success"),
+			readFailureMetric:          generatorMetrics.NewCountMetric("read_failure"),
+			operatorSuccessMetrics:     make(map[core.OperatorID]metrics.CountMetric),
+			operatorFailureMetrics:     make(map[core.OperatorID]metrics.CountMetric),
+			requiredReadPoolSizeMetric: generatorMetrics.NewGaugeMetric("required_read_pool_size"),
+			optionalReadPoolSizeMetric: generatorMetrics.NewGaugeMetric("optional_read_pool_size"),
+		},
+	}
+}
+
+// Start begins a blob reader goroutine.
+func (r *BlobReader) Start() {
+	r.waitGroup.Add(1)
+	ticker := time.NewTicker(r.config.ReadRequestInterval)
+	go func() {
+		defer r.waitGroup.Done()
+		for {
+			select {
+			case <-(*r.ctx).Done():
+				err := (*r.ctx).Err()
+				if err != nil {
+					r.logger.Info("blob r context closed", "err:", err)
+				}
+				return
+			case <-ticker.C:
+				r.randomRead()
+			}
+		}
+	}()
+}
+
+// randomRead reads a random blob.
+func (r *BlobReader) randomRead() {
+	metadata := r.blobsToRead.GetNext()
+	if metadata == nil {
+		// There are no blobs that we are required to read.
+		return
+	}
+
+	r.metrics.requiredReadPoolSizeMetric.Set(float64(r.blobsToRead.Size()))
+
+	ctxTimeout, cancel := context.WithTimeout(*r.ctx, r.config.FetchBatchHeaderTimeout)
+	defer cancel()
+
+	start := time.Now()
+	batchHeader, err := r.chainClient.FetchBatchHeader(
+		ctxTimeout,
+		gcommon.HexToAddress(r.config.EigenDAServiceManager),
+		metadata.BatchHeaderHash[:],
+		big.NewInt(int64(0)),
+		nil)
+	if err != nil {
+		r.logger.Error("failed to get batch header", "err:", err)
+		r.metrics.fetchBatchHeaderFailure.Increment()
+		return
+	} else {
+		end := time.Now()
+		duration := end.Sub(start)
+		r.metrics.fetchBatchHeaderMetric.ReportLatency(duration)
```

Comment:
These 3 lines can be simplified as `r.metrics.fetchBatchHeaderMetric.ReportLatency(time.Since(start))`

Reply:
change made